### DISCUSSION TITLE
how to cross compile zmq.go

### DISCUSSION THREAD

**Discussion by iceteas** [lf-edge/ekuiper#1702] [General]
I specified the path to libzmq.pc for the customized arm version, which also appears to be working, but an error was reported anyway
![image](https://user-images.githubusercontent.com/12604419/228160728-1fcb1c91-3ea4-4ac7-9bcc-b47fe64f4bfb.png)

**Comment by ngjaying** [comment]
There is a prebuilt plugin to be installed from the web console. You could try to install: https://ekuiper.org/docs/en/latest/operation/manager-ui/overview.html. And then in the "extensions" tab, install the zmq plugin.

**Reply by iceteas** [reply]
I didn't use docker and just wanted to cross-compile into aarch64 to run native

**Reply by superrxan** [reply]
* Why there is a `permission denied` error ? can you build out kuiperd ?

* Do you run offically released kuiperd? if so , we have ZMQ plugin offically released, you can get by these steps

1.   get the address of plugins
 
  ``` shell
       curl http://localhost:9081/plugins/sources/prebuild
  ```
    sample output

  ``` json
  {
    "file": "https://packages.emqx.net/kuiper-plugins/1.8.1/sources/debian/file_arm64.zip",
    "zmq": "https://packages.emqx.net/kuiper-plugins/1.8.1/sources/debian/zmq_arm64.zip"
  }  
  ```
2.   create new plugins

``` shell
 POST http://localhost:9081/plugins/sources
{
  "name":"zmq",
  "file":"https://packages.emqx.net/kuiper-plugins/1.8.1/sources/debian/zmq_arm64.zip"
}
```

**Reply by iceteas** [reply]
![image](https://user-images.githubusercontent.com/12604419/228213077-3f014207-f2e3-4b5e-8c4b-e0da32f0e9f6.png)

**Reply by ngjaying** [reply]
The prebuilt plugin must be pair with the prebuilt eKuiper downloaded from Github releases.
